securedrop_client/gui/base/inputs.py

```diff
@@ -30,20 +28,15 @@ def __init__(self, parent: QDialog) -> None:
         self.parent = parent
         super().__init__(self.parent)
 
-        self.visibleIcon = load_icon("eye_visible.svg")
-        self.hiddenIcon = load_icon("eye_hidden.svg")
-
         self.setEchoMode(QLineEdit.Password)
-        self.togglepasswordAction = self.addAction(self.hiddenIcon, QLineEdit.TrailingPosition)
-        self.togglepasswordAction.triggered.connect(self.on_toggle_password_Action)
         self.password_shown = False
 
     def on_toggle_password_Action(self) -> None:
+        state = QCheckBox()
         if not self.password_shown:
             self.setEchoMode(QLineEdit.Normal)
             self.password_shown = True
-            self.togglepasswordAction.setIcon(self.visibleIcon)
+            state.isChecked()
```

Comment:
At first glance, I don't understand what this additional checkbox does. Caveat: it's a quick read-through and that doesn't mean anything it wrong with it!